tests/hilti/types/bytes/iterator.hlt

```diff
@@ -18,15 +18,6 @@ assert *i++ == '1';
 assert *i++ == '2';
 assert *++i == '4';
 
-assert i > j;
```

Comment:
Did we lose the comparison operators? They would be good to keep, so that one can build custom loops.

Reply:
This is the only place we ever exercised this kind of operator for operators, and we e.g., do not support it for `rt::Vector` iterators either. General STL-type iterators are typically only compared for inequality (e.g., `it != container.end()`), but if we need iterators to be ordered as well I can add the operators back. In that case lets come up with some code which actually exercises that property.

Reply:
I added the ordering of iterators into `rt::Bytes` back (along with the previously removed test case), and also opened #363 for adding the same property to iterators into `rt::Vector`. As `rt::List` is not index-based I didn't add this property to its iterators, but if it turns out to be important we can change its iterators as well to hold the needed internal state.